eng/docker/libraries-sdk.linux.Dockerfile

```diff
@@ -8,7 +8,7 @@ WORKDIR /repo
 COPY . .
```

Comment:
On line 2: the current tag for the target sdk image is `3.0.100` which is probably a bit too old. Unless the build argument is being overridden by the build script this should probably be updated?

Reply:
I probably agree that we need to update the base image version, but I think it should be done separately since this PR only deals with agent pool.